gha/apt-dependencies/action.yaml

```diff
@@ -29,22 +30,24 @@ runs:
       with:
         path: |
           ~/apt_cache
-        key: ${{ runner.os }}-apt-${{ input.cache-version }}-${{ hashFiles(github.action_path + '/apt-packages.txt') }}
+        key: ${{ runner.os }}-apt-${{ inputs.cache-version }}-${{ hashFiles(format('{0}{1}', github.action_path, '/apt-packages.txt')) }}
         restore-keys: |
-          ${{ runner.os }}-apt-${{ input.cache-version }}-
+          ${{ runner.os }}-apt-${{ inputs.cache-version }}-
 
     - name: Install APT Dependencies
+      shell: bash
       env:
         CACHE_HIT: ${{steps.cache-apt-deps.outputs.cache-hit}}
-        APT_PACKAGES_FILE_PATH: ${{github.action_path + '/apt-packages.txt'}}
+        APT_PACKAGES_FILE_PATH:  ${{ format('{0}{1}', github.action_path, '/apt-packages.txt') }}
       run: |
         if [[ "$CACHE_HIT" != 'true' ]]; then
         sudo apt-get -y update
         fi
-        APT_PACKAGES=$(grep -v '^#' "${APT_PACKAGES_FILE_PATH}" | xargs echo -n)
+        APT_PACKAGES=$(grep -v '^#' "${{env.APT_PACKAGES_FILE_PATH}}" | xargs echo -n)
```

Comment:
The `${{ }}` syntax gets substituted by github's workflow engine. I used `${ }` because I expected bash to handle substituting the var. Isn't `$APT_PACKAGES_FILE_PATH` available here?

If using `env:` to add vars is not working, then several things in the Makefile will be broken in later steps. So, I think we need to figure that out before continuing to much farther into the Makefile.

Reply:
https://docs.github.com/en/actions/learn-github-actions/environment-variables

Reply:
Ok, I'll scan the file differentiating between those two syntaxes and make sure they link up correctly and run some tests.

Reply:
Ok, so here is a test output for using `${ }` vs `${{ }}` :
![single-vs-double](https://user-images.githubusercontent.com/92544319/143068179-e5ad6fef-cfc3-49e0-b332-28359ede3dd3.png)

The single brackets use the pure string whereas the double brackets appear to format it as the actual path. So, this is good to know.

Reply:
When you use the var, bash will expand `~` to `/home/runner`, so I thought they would be effectively the same... apparently not. We need the `env:` block to work..., so I have another suggestion...